packages/lesswrong/server/resolvers/commentResolvers.ts

```diff
@@ -192,4 +200,82 @@ augmentFieldsDict(Comments, {
       return (post && post.contents && post.contents.version) || "1.0.0";
     },
   },
+  doppelComments: {
+    resolveAs: {
+      type: '[DoppelComment]',
+      resolver: async (dbComment, context: ResolverContext) => {
+        const user = await Users.findOne({ _id: dbComment.userId });
+        if (!user) return []
+        if (user.karma < 1e4) return []
+        const doppelComments = await runFragmentQuery({
+          fragmentName: 'DoppelCommentsFragment',
+          terms: { view: 'doppelCommentsForComment', commentId: dbComment._id },
+          collectionName: 'DoppelComments',
+        })
+
+
+        if (doppelComments.length < 2) {
+          // We'll make 'em for next time
+          void [createDoppelComment(dbComment, user), createDoppelComment(dbComment, user)]
+        }
+        return doppelComments
+      }
+    }
+  }
 });
+
+const createDoppelComment = async (dbComment: DbComment, user: DbUser) => {
+  const [interestingGwernComment, eightShortStudies, topUserComment, topUserPost] = await Promise.all([
+    Comments.findOne({ _id: 'hxMNNJRF6o644aNTa'}),
+    Posts.findOne({_id: 'gFMH3Cqw4XxwL69iy'}),
+    Comments.findOne({ userId: dbComment.userId }, { sort: { 'baseScore': -1 }, limit: 1, projection: { '_id': 1 } }),
+    Posts.findOne({ userId: dbComment.userId }, { sort: { 'baseScore': -1 }, limit: 1, projection: { '_id': 1 } }),
+  ])
+  if (!interestingGwernComment || !eightShortStudies || !topUserComment || !topUserPost) {
+    const docIdentifiers = ["interestingGwernComment", "eightShortStudies", "topUserComment", "topUserPost"]
+    const failedToFind = [interestingGwernComment, eightShortStudies, topUserComment, topUserPost].flatMap((doc, i) => !doc ? [docIdentifiers[i]] : [])
+    captureMessage(`Failed to find all necessary documents for doppel comments on ${dbComment._id}. Failed to find: ${failedToFind.join('\n')}`, Severity.Warning)
+    return
+  }
+  const messageHistory = await constructMessageHistory(
+    '',
+    [interestingGwernComment, topUserComment].map(comment => comment._id),
+    [eightShortStudies, topUserPost].map(post => post._id),
+    user,
+    createAdminContext(),
+    dbComment.parentCommentId ?? undefined,
+    dbComment.postId ?? undefined,
+  )
+
+  const client = getAnthropicPromptCachingClientOrThrow()
+
+  let res = null
+  try {
+    res = await client.messages.create({
+      model: 'claude-3-5-sonnet-20240620',
+      max_tokens: 2000,
+      system: "The assistant is in CLI simulation mode, and responds to the user's CLI commands only with the output of the command.",
+      messages: messageHistory,
+      stop_sequences: ['---'],
```

Comment:
why custom `stop_sequences`?

Reply:
This is our separator between documents in the prompt. This ensures we don't e.g. start generating a new comment. Oli doesn't want it on the `autocompleteEndpoint` because he sometimes likes generating the replies

Reply:
ack